nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/CommonClassLoaderServiceImpl.java

```diff
@@ -189,87 +178,57 @@ private static String urlsToClassPath(Stream<URL> urls) {
         return urls.map(FileUtils::toFile).map(File::getAbsolutePath).collect(Collectors.joining(File.pathSeparator));
     }
 
-    private List<File> findDerbyClient() {
-        final String DERBY_HOME_PROP = "AS_DERBY_INSTALL";
-        StartupContext startupContext = env.getStartupContext();
-        Properties arguments = null;
+    private List<File> findDerbyJars(File installDir) {
+        Path derbyHome = getDerbyDir(installDir);
+        LOG.log(CONFIG, "Using derby home: {0}", derbyHome);
 
-        if (startupContext != null) {
-            arguments = startupContext.getArguments();
+        final File derbyLib = derbyHome.resolve("lib").toFile();
+        if (!derbyLib.exists()) {
+            LOG.info(KernelLoggerInfo.cantFindDerby);
+            return Collections.emptyList();
         }
 
-        String derbyHome = null;
-
-        if (arguments != null) {
```

Comment:
Is it OK now to ignore any `DERBY_HOME_PROP` property in `StartupContext`? Is it not used anywhere?

Reply:
Seems it is set just in `ACCStartupContext`. So you are probably right, I should use better strategy for these changes and always check following changes in my branch which probably fixed that. On the other hand no test failed, GF is quite overengineered here. Or underengineered with all those dependencies crossing paths ...

Reply:
I started understanding why this ... all those values have single source, however BootstrapKeys are not available on classpath everywhere. 
But still, convention of the StartupContext is to use "JVM options like" keys, not ENV like keys.